plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2287,13 +2287,28 @@ public void validateScan(ConnectorSession session, ConnectorTableHandle tableHan
         Map<ConnectorExpression, ProjectedColumnRepresentation> columnProjections = projectedExpressions.stream()
                 .collect(toImmutableMap(Function.identity(), HiveApplyProjectionUtil::createProjectedColumnRepresentation));
 
-        // No pushdown required if all references are simple variables
+        HiveTableHandle hiveTableHandle = (HiveTableHandle) handle;
+        // all references are simple variables
         if (columnProjections.values().stream().allMatch(ProjectedColumnRepresentation::isVariable)) {
-            return Optional.empty();
+            Set<ColumnHandle> projectedColumns = ImmutableSet.copyOf(assignments.values());
+            if (hiveTableHandle.getProjectedColumns().isPresent()
+                    && hiveTableHandle.getProjectedColumns().get().equals(projectedColumns)) {
+                return Optional.empty();
+            }
+            return Optional.of(new ProjectionApplicationResult<>(
+                    hiveTableHandle.withProjectedColumns(projectedColumns),
+                    projections,
+                    assignments.entrySet().stream()
```

Comment:
Aren't you just making a strict copy of `assignments` here? Why not just use `assignments`? Or did you mean to filter out some netries?

Reply:
I am making a copy, but I need to convert `Map<String, ColumnHandle> assignments` to `List<Assignment>`.
Let me know if there is a better to do this.

Reply:
Oh I see it know. It I do not see a much nicer way to do this. What would be more readable to me is to extract this as a named `List<Assignment> assignmentsList` variable.

Reply:
updated to extract `List<Assignment> assignmentsList` variable